pkg/middleware/middleware.go

```diff
@@ -300,7 +300,7 @@ func AddDefaultResponseHeaders() macaron.Handler {
 
 // AddSecurityHeaders adds various HTTP(S) response headers that enable various security protections behaviors in the client's browser.
 func AddSecurityHeaders(w macaron.ResponseWriter) {
-	if (setting.Protocol == setting.HTTPS || setting.Protocol == setting.HTTP2) && setting.StrictTransportSecurity {
+	if setting.Protocol == setting.HTTP2 && setting.StrictTransportSecurity {
```

Comment:
Hm, this doesn't look right. We want to go from:

|                   | HTTP | HTTPS | HTTP2 |
| ---------- | ------ | ------- | ------- |
| HSTS ON  | N         | Y          | Y           |
| HSTS OFF | N        |  N          | N          |

to

|                   | HTTP | HTTPS | HTTP2 |
| ---------- | ------ | ------- | ------- |
| HSTS ON  | Y        | Y          | Y           |
| HSTS OFF | N        |  N          | N          |

But this looks like:

|                   | HTTP | HTTPS | HTTP2 |
| ---------- | ------ | ------- | ------- |
| HSTS ON  | N         | N          | Y           |
| HSTS OFF | N        |  N          | N          |

Am I missing something?

Reply:
In case I'm getting this right, the fix would be

```suggestion
	if setting.StrictTransportSecurity {
```